rich/logging.py

```diff
@@ -70,8 +83,19 @@ def __init__(
         super().__init__(level=level)
         self.console = console or get_console()
         self.highlighter = highlighter or self.HIGHLIGHTER_CLASS()
+
+        for level, level_name in self.__class__.levels.items():
+            logging.addLevelName(level, level_name)
+
+        level_width = self.__class__.level_width or max(
+            len(level_name) for level_name in logging._levelToName.values()
```

Comment:
The use of `logging._levelToName` might be still a problem based on your comment. But I don't find a better way to walk through all level names, including the default ones and the ones defined in `levels` by `logging.addLevelName`.

Looks like `_levelToName` is where the levels and their names stored: https://github.com/python/cpython/blob/c12afa92b0db6f017e479b41f95d75bac7b59850/Lib/logging/__init__.py#L142

Let me know if you have a better way. Or the other idea is that we just walk through the levels defined in the subclass, instead of all available levels. This means all longer names (than `level_width`) should be shortened/renamed in `levels` if one doesn't want them to be cut.